### PR TITLE
Fix sts webhook error

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 
Fix the error which would cause panic for sts webhook when sts updated strategy is not rolling updated.

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
Fix a bug which would cause panic in statefulset webhook when update strategy of `StatefulSet` is not `RollingUpdate`
```

### CODE REVIEW

File: pkg/webhook/statefulset/statefulset.go [hunk 1]
```diff
@@ -110,12 +110,11 @@ func (sc *StatefulSetAdmissionControl) AdmitStatefulSets(ar *admission.Admission
...
 	}
 
-	setPartition := *(stsPartition)
-	if setPartition > 0 && setPartition <= int32(partition) {
+	if stsPartition != nil && *stsPartition > 0 && *stsPartition <= int32(partition) {
 		klog.Infof("statefulset %s/%s has been protect by partition %s annotations", namespace, name, partitionStr)
 		return util.ARFail(errors.New("protect by partition annotation"))
 	}
-	klog.Infof("admit statefulset %s/%s update partition to %d, protect partition is %d", namespace, name, setPartition, partition)
+	klog.Infof("admit statefulset %s/%s update partition to %d, protect partition is %d", namespace, name, *stsPartition, partition)
```
Comment (DanielZhangQD) [inline pkg/webhook/statefulset/statefulset.go:117]: `stsPartition` is possible to be nil here.

File: pkg/webhook/statefulset/statefulset.go [hunk 2]
```diff
@@ -125,12 +124,18 @@ func getStsAttributes(data []byte, apiVersion string) (*metav1.ObjectMeta, *int3
...
 
 	set := apps.StatefulSet{}
 	if _, _, err := deserializer.Decode(data, nil, &set); err != nil {
 		return nil, nil, err
 	}
-	return &(set.ObjectMeta), set.Spec.UpdateStrategy.RollingUpdate.Partition, nil
+	if set.Spec.UpdateStrategy.RollingUpdate != nil {
+		return &(set.ObjectMeta), set.Spec.UpdateStrategy.RollingUpdate.Partition, nil
+	}
+	return &(set.ObjectMeta), nil, nil
```
Comment (cofyc) [inline pkg/webhook/statefulset/statefulset.go:140]: the code seems identical no matter what apiVersion is?

**Comment by cofyc** [comment]
Suggested release note:

```
Fix a bug which would cause panic in statefulset webhook when update strategy of `StatefulSet` is not `RollingUpdate`
```

https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md

File: pkg/webhook/statefulset/statefulset.go (hunk 1 above)
Comment (Yisaer) [reply pkg/webhook/statefulset/statefulset.go]: updated

File: pkg/webhook/statefulset/statefulset.go (hunk 2 above)
Comment (Yisaer) [reply pkg/webhook/statefulset/statefulset.go:131]: updated.

**Pull request merged**